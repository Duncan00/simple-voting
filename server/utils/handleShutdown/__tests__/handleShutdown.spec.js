'use strict';

const handleShutdown = require('..');

describe('handleShutdown', () => {
	test('should listen to shutdown signal and setup', () => {
		const server = {};
		const app = {};
		const redis = {};
		const logger = {};
		process.on = jest.fn();

		handleShutdown({server, app, redis, logger});

		expect(process.on.mock.calls).toMatchSnapshot('process.on');
	});
});
