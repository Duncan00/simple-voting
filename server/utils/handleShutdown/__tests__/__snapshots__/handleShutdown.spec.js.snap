// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`handleShutdown should listen to shutdown signal and setup: process.on 1`] = `
Array [
  Array [
    "SIGINT",
    [Function],
  ],
  Array [
    "SIGTERM",
    [Function],
  ],
]
`;
